### ISSUE TITLE
[button] axe-core 4.2.0 failings on `nested-interactive

### ISSUE THREAD

**Issue by daKmoR** [ing-bank/lion#1353]
## Expected behavior

when updating axe-core to lastest version (4.2.0) accessibility checks should pass

## Actual Behavior

Test like button fail with

```
lion-button > accessibility > passes a11y audit
      Error: Accessibility Violations
      ---
      Rule: nested-interactive
      Impact: serious
      Ensure interactive controls are not nested (https://dequeuniversity.com/rules/axe/4.2/nested-interactive?application=axeAPI)

      Issue target: lion-button[text=""]
      Context: <lion-button text="" tabindex="0" role="button" type="submit" aria-disabled="false">foo<button tabindex="-1" aria-hidden="true" slot="_button" type="submit"></button></ing-button>
      Fix any of the following:
        Element has focusable descendants
      ---
```

## Additional context

on latest master

**Comment by tlouisse** [comment]
Maybe we can solve it by adding [role=presentation] on the nested button

**Comment by daKmoR** [comment]
I sort of was surprised by it as though we only spawn the button on click... 

will test the role 👍

**Comment by daKmoR** [comment]
ok, role did not work out 😅

opened an issue with axe-core as I think it should be ok... let's see what they say
https://github.com/dequelabs/axe-core/issues/2906

**Comment by tlouisse** [comment]
I think we had the native button (without `[role="button"][tabindex=0]` wrapper) in the very beginning.
Problems you would bump into:
- no styling of pseudo classes like `:focus` etc on host
- click handlers will have unexpected targets

There's probably a really easy way to solve this. The only reason we have the _nativeButtonNode is to detect the form on connecredCallback. We just have to use the same strategy as for form submission here: spawn the __submitAndResetHelperButton for an instant of time:

```js
  /**
   * @private
   */
  __setupSubmitAndResetHelperOnConnected() {
    this._form = this._nativeButtonNode.form;
    ...
  }
```
=> 
```js
  /**
   * @private
   */
  __setupSubmitAndResetHelperOnConnected() {
      this._form.appendChild(this.__submitAndResetHelperButton);
      this._form = this.__submitAndResetHelperButton.form;
      this._form.removeChild(this.__submitAndResetHelperButton);
      ...
  }
```
And then we can just remove `_nativeButtonNode`, including type delegation (it's superfluous atm already)

Also, amongst the list of features we should probably als mention the ability to add shadow dom (it's somewhat reflected by the icon slots), because without it we wouldn't have been able to create SelectRichInvoker etc.
Also, we should add a note that native buttons should always be used wherever they make sense.

For v1, we might want to create a clean `LionButton` and a `LionSubmitButton` extension (latter would contain the 'necessary 
 dirt')

**Comment by daKmoR** [comment]
to get implicit form submission we need a permanent button as described in our docs https://lion-web.netlify.app/components/interaction/button/features/#considerations 

> ## Native button & implicit form submission
>Flashing the button in the way we do solves almost all issues except for one. One of the specs of W3C is that when you >have a form with multiple inputs, pressing enter while inside one of the inputs only triggers a form submit if that form has >a button of type submit.
>To get this particular implicit form submission to work, having a native button in our <lion-button> is a hard requirement. >Therefore, not only do we flash a native button on the form to delegate <lion-button> trigger to <button> and thereby >trigger form submission, we also add a native button inside the <lion-button> whose type property is synchronized with >the type of the <lion-button>.

**Comment by tlouisse** [comment]
Ah right, how could I forget that one. Yeah, the only way we can do it then is by adding the button element outside of our host (as a sibling or last child of `_form`), but this feels a bit tricky.

**Comment by daKmoR** [comment]
New Plan:

## Phase 1:
- Created native button outside of the inner HTML
```html
<form>
  <!-- ...  -->
  <lion-button></lion-button>
  <!-- ...  -->
  <button type="submit" style="visually-hidden-css-styles" generated-button-for-implicit-form-submit></button>
</form>
```
- Make sure only ONE outside button gets created no matter how many lion buttons are there
- Check if a mutation observer is needed on the native button? e.g. if removed, readd it back
- pass the axe-core test

## Phase 2 => breaking change:

- Split into two web components
- `<lion-button>` not having any of those "hack" for adding a native button (or spawning a temporary button for clicking)
- `<lion-button>` will become type="button" by default
- `<lion-button-submit>` explicit decision for needing submit and it hacks
- `<lion-button-reset>` explicit decision for needing reset and it hacks
- => before starting this let's refine it a little more
